cli/connect/console.go

```diff
@@ -0,0 +1,352 @@
+	console.historyLines, err = common.GetLastNLines(console.historyFilePath, MaxHistoryLines)
+	if err != nil {
+		return
+	}
+}
+
+func setTitle(console *Console) {
+	if console.title != "" {
+		return
+	}
+
+	titleRaw, err := console.Eval(getTitleFuncBody)
+	if err == nil {
+		if title, ok := titleRaw.(string); ok {
+			console.title = title
+		}
+	}
+
+	if console.title == "" {
+		console.title = console.connOpts.Address
+	}
+}
+
+func setPrefix(console *Console) {
+	console.prefix = fmt.Sprintf("%s> ", console.title)
+
+	livePrefixIndent := len(console.title)
+	if livePrefixIndent > MaxLivePrefixIndent {
+		livePrefixIndent = MaxLivePrefixIndent
+	}
+
+	console.livePrefix = fmt.Sprintf("%s> ", strings.Repeat(" ", livePrefixIndent))
+
+	console.livePrefixFunc = func() (string, bool) {
+		return console.livePrefix, console.livePrefixEnabled
+	}
+}
+
+func getExecutor(console *Console) (prompt.Executor, error) {
+	var executeFunc func(console *Console, in string) string
+	switch {
+	case console.protocol == PlainTextProtocol:
+		executeFunc = plainTextExecute
+	case console.protocol == BinaryProtocol:
+		executeFunc = binaryExecute
+	default:
+		return nil, fmt.Errorf("Unknown protocol: %s", console.protocol)
+	}
+
+	executor := func(in string) {
+		console.input += in + " "
+
+		if !inputIsCompleted(console.input, console.luaState) {
+			console.livePrefixEnabled = true
+			return
+		}
+
+		handleSetOutput(console, console.input)
+
+		if console.historyFilePath != "" {
+			common.AppendStringToFile(console.historyFilePath, console.input+"\n")
+		}
+
+		data := executeFunc(console, console.input)
+
+		fmt.Printf("%s\n", data)
+
+		console.input = ""
+		console.livePrefixEnabled = false
+	}
+
+	return executor, nil
+}
+
+func inputIsCompleted(input string, luaState *lua.LState) bool {
+	var err error
+
+	_, err = luaState.LoadString(input)
+	if err == nil || !strings.Contains(err.Error(), "at EOF") {
+		// valid Lua code or a syntax error not due to
+		// an incomplete input
+		return true
+	}
+
+	if _, err = luaState.LoadString(fmt.Sprintf("return %s", input)); err == nil {
+		// certain obscure inputs like '(42\n)' yield the
+		// same error as incomplete statement
+		return true
+	}
+
+	return false
+}
+
+func getCompleter(console *Console) (prompt.Completer, error) {
+	switch {
+	case console.protocol == PlainTextProtocol:
+		return getPlainTextCompleter(console), nil
+	case console.protocol == BinaryProtocol:
+		return getBinaryCompleter(console), nil
+	}
+
+	return nil, fmt.Errorf("Unknown protocol: %s", console.protocol)
+}
+
+func handleSetOutput(console *Console, in string) {
+	inWords := strings.Fields(in)
+
+	if len(inWords) != 3 {
+		return
+	}
+
+	if inWords[0] != "\\set" || inWords[1] != "output" {
+		return
+	}
+
+	if ConsoleOutputMode(inWords[2]) == ConsoleYAMLOutput {
+		console.outputMode = ConsoleYAMLOutput
+		return
+	}
+
+	if ConsoleOutputMode(inWords[2]) == ConsoleLuaOutput {
+		console.outputMode = ConsoleLuaOutput
+		return
```

Comment:
This  return is not necessary

Reply:
This return makes this `if` block the same as `if` block above. Moreover, it allows to add any code below without changing this block and prevents future mistakes.